### PR TITLE
Remove unused exception handling

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39486 Add @rpc.functions.async_execution for rpc.remote
* #39485 Make @rpc.functions.async_execution processing generic
* #38398 Explicitly decref in UnpickledPythonCall dtor
* #39489 Run clang-format for RPC code
* **#39469 Remove unused exception handling**
* #39267 Add rpc.functions.async_execution decorator for TorchScript functions

The exception has already been handled in `_run_function` in
`internal.py`. Hence, it is not necessary to do it again in
`request_callback_impl.cpp`.

Differential Revision: [D21868480](https://our.internmc.facebook.com/intern/diff/D21868480)

### CODE REVIEW

File: torch/csrc/distributed/rpc/request_callback_impl.cpp [hunk 1]
```diff
@@ -361,23 +361,13 @@ void RequestCallbackImpl::processRpc(
...
-        ownerRRef->setValue(std::move(py_ivalue));
-      } catch (py::error_already_set& e) {
-        // py::error_already_set requires GIL to destruct, take special care.
-        ownerRRef->setError(e.what());
+      {
         py::gil_scoped_acquire acquire;
-        e.restore();
-        PyErr_Clear();
-      } catch (std::exception& e) {
-        ownerRRef->setError(e.what());
```
Comment (pritamdamania87) [inline torch/csrc/distributed/rpc/request_callback_impl.cpp:379]: [_run_function](https://github.com/pytorch/pytorch/blob/master/torch/distributed/rpc/internal.py#L149) doesn't seem to set the error on the ownerRRef though? Don't we still need this?

Comment (mrshenli) [reply torch/csrc/distributed/rpc/request_callback_impl.cpp:379]: This was never executed, as exceptions thrown by the python udf is wrapped in a RemoteException object, which will be re-thrown when calling `to_here()`.

Comment (mrshenli) [reply torch/csrc/distributed/rpc/request_callback_impl.cpp:379]: Hmm, FWIW, those cbs also all have their own try-catch and doesn't seem would throw an error.

Comment (mrshenli) [reply torch/csrc/distributed/rpc/request_callback_impl.cpp:379]: Let me move this PR out of this stack, as we are unsure about the impact here. And will add some more tests later.

**Pull request closed without merging**